docs/troubleshoot/app-framework/appserver-error-codes.md

```diff
@@ -1800,291 +1800,348 @@ The following error message codes may appear on the app-server log. Use the foll
+  and throws _errorMessage_ in the catch block.
 
   **Action:**
 
-  TODO
+  No action is needed, but the _errorMessage_ may be needed to debug
 
 
 
-### ZWED0071W":"Unexpected error on server %s:%s. E=%s. Stack trace follows.\n%s
+### ZWED0071W
+
+  Unexpected error on server _ipAddress_: _port_ Error=_errorMessage_ stack trace Exception
 
   **Reason:**
 
-  TODO
+  When we get an unexpected error on the server with _ipAddress_ and _port_.
 
   **Action:**
 
-  TODO
+  No action needed unless user is experiencing an interruption in server, then send stack to developers
+
 
 
+### ZWED0072W
 
-### ZWED0072W":"Could not stop manager, error=
+  Could not stop manager due to error _errorMessage_
 
   **Reason:**
 
-  TODO
+  If the server manager is unable to stop due to any reason it will throw an exception with an _errorMessage_.
 
   **Action:**
 
-  TODO
+  Review the _errorMessage_ and see if there is something you can do to fix the error
 
 
 
-### ZWED0073W":"No server returned for group=
+### ZWED0073W
+
+  No server returned for group=_group_
 
   **Reason:**
 
-  TODO
+  If no server returned for _group_ element from _groupingConfig_ array
+  then it will throw this warning message.
 
   **Action:**
 
-  TODO
+  Review the warning message and check for the _group_ element to fix the error
 
 
+### ZWED0074W
 
-### ZWED0074W":"Unknown default behavior=%s
+  Unknown default behavior=_defaultBehavior_
 
   **Reason:**
 
-  TODO
+  If the server is not in a microservice or appserver then it logs a default
+  switch case warning message as the unknown _defaultBehavior_.
 
   **Action:**
 
-  TODO
+  No action is needed, but the warning may be needed to debug
 
 
 
-### ZWED0075W":"Services in plugin=%s war grouping skipped. Plugin missing or already grouped
+### ZWED0075W
+
+  Services in plugin=_plugin[i]_ war grouping skipped. Plugin missing or already grouped
 
   **Reason:**
 
-  TODO
+  If element from _plugins_ array, does not contain a compatible service (data service plugin/war service)
+  then the element is skipped and not added to a new array.
 
   **Action:**
 
-  TODO
+  Review the warning message and check for the _plugins_ array element to fix the error
+
 
 
+### ZWED0076W
 
-### ZWED0076W":"Skipping invalid plugin group=
+  Skipping invalid plugin group=_plugins_
 
   **Reason:**
 
-  TODO
+  If the _plugins_ is not an array and the size is less than zero, then it will log a
+  warning message in the else block.
 
   **Action:**
 
-  TODO
+  Make sure _plugins_ should be an array of size greater than zero.
+
 
 
+### ZWED0077W
 
-### ZWED0077W":"Could not extract war for service=%s, error=
+  Could not extract war for service=_key-value_, error=_errorMessage_
 
   **Reason:**
 
-  TODO
+  If the service with the _key-value_ pair is unable to extract the war file then it throws the _errorMessage_
 
   **Action:**
 
-  TODO
+  Check if your war file is in a valid format or exists.
```

Comment:
Can you change it to:
Check if **the war file exists and has a valid format.**

Reply:
Updated requested changes as per suggestions. Please have a look.